website/content/docs/guides/unseal/pkcs11/utimaco.mdx

```diff
@@ -0,0 +1,394 @@
+  over the network or via PCIe)
+- An admin account key is available to manage a PKCS#11 slot, or a PKCS#11
+  slot and user have already been initialized
+
+Utimaco HSMs come with several command line tools that will be used throughout
+this guide. Ensure the following are available:
+
+- `csadm` - HSM administration
+- `cxitool` - Manage and interact with cryptographic functionality
+- `p11tool2` - PKCS#11 specific functions, not to be confused with GNU `p11tool`
+
+To begin, set the `CRYPTOSERVER` environment variable to point `csadm` and `cxitool` at the HSM:
+
+```shell-session
+$ export CRYPTOSERVER=288@194.168.4.107
+```
+
+Depending on your device, use:
+
+- `CRYPTOSERVER=<PORT>:<ADDRESS>` to connect over the network
+- `CRYPTOSERVER=/dev/cs2.<INDEX>` to connect to an attached PCIe card
+
+Verify that the HSM is reachable:
+
+```shell-session
+$ csadm GetState
+
+mode      = Operational Mode
+state     = INITIALIZED (0x00100004)
+
+-- more output --
+```
+
+Then set `CS_AUTH_KEYS` and download the HSM's public key:
+
+```shell-session
+$ export CS_AUTH_KEYS=/etc/utimaco/HSM_AUTH.kak
+$ csadm GetHSMAuthKey > "$CS_AUTH_KEYS"
+```
+
+This key is required to authenticate against the HSM in combination with your
+user's private key.
+
+Finally, create a configuration file for `p11tool2`. This file can be placed
+at an arbitrary location, but you must point at it by setting an environment
+variable.
+
+<Tabs>
+
+<TabItem value="u.trust GP HSM" heading="utrust-p11tool">
+
+```shell-session
+$ export CS_PKCS11_R3_CFG=/etc/utimaco/cs_pkcs11_R3.cfg
+```
+
+Then place _at least_ the following at /etc/utimaco/cs_pkcs11_R3.cfg:
+
+</TabItem>
+
+<TabItem value="CryptoServer CP5" heading="cryptoserver-p11tool">
+
+```shell-session
+$ export CS_PKCS11_R2_CFG=/etc/utimaco/cs_pkcs11_R2.cfg
+```
+
+Then place _at least_ the following at /etc/utimaco/cs_pkcs11_R2.cfg:
+
+</TabItem>
+
+</Tabs>
+
+```ini
+[Global]
+
+# No required options here, but the section must exist.
+
+[CryptoServer]
+
+# This is equal to $CRYPTOSERVER
+Device = 288@194.168.4.107
+```
+
+There are many more options available, and the above is only the minimum
+required configuration. Refer to Utimaco's official PKCS#11 guide (delivered
+together with your HSM) for an exhaustive list.
+
+This will make PKCS#11 slots 0-9 available by default.
+
+## Initializing a PKCS#11 Slot
+
+:::info
+
+If you have already initialized a PKCS#11 slot, you may skip to the [next
+section](#creating-the-key-material)
+
+:::
+
+Next, initialize a PKCS#11 slot and assign a Security Officer PIN + User PIN to
+it. For the purpose of this guide, we use slot number 1 and a dummy PIN value of
+`4321` for the Security Officer, and `1234` for the User.
+
+<Tabs>
+
+<TabItem value="u.trust GP HSM" heading="utrust-slot">
+
+Initialize the slot using an administrator account and set the SO PIN:
+
+```shell-session
+$ p11tool2 Slot=1 Login=ADMIN,/etc/utimaco/ADMIN.key InitToken=temp
```

Comment:
is `p11tool2` from utimaco? With the other documentation we used `pkcs11-tool`. I think if we could use the same tool in all docs it would be nice, but if the `p11tool2` comes directly from Utimaco than that's fine too.